src/help/git.ts

```diff
@@ -13,10 +13,19 @@ export function clone(repositoryUrl: string, targetDir: string) {
     shell.run(`git clone git@${sshRepositoryUrl}.git ${targetDir}`);
   } else {
     const gitHubToken = process.env.GITHUB_TOKEN;
-    if (!gitHubToken) {
-      throw new Error('GITHUB_TOKEN env variable is required when GITHUB_USE_SSH env variable is not used');
+    if (gitHubToken) {
+      shell.run(`git clone https://${gitHubToken}@${repositoryUrl}.git ${targetDir}`);
+    } else {
```

Comment:
If I read this correctly `GITHUB_TOKEN` is present even in GHE workflows: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Reply:
So I think the logic goes something like this

```
function getToken(isGHE: boolean) {
  if (isGhE()) {
    return process.env.GH_ENTERPRISE_TOKEN ?? process.env.GITHUB_TOKEN; 
  }
  return process.env.GITHUB_TOKEN;
}
```

```
const isGHE = detectGHE();
const token = getToken(isGHE);
if (!token) throw Error;
shell.run(`git clone https://${token}@${repositoryUrl}.git ${targetDir}`);
```